documentation/jetty-documentation/src/main/asciidoc/programming-guide/server/http/server-http-handler-implement.adoc

```diff
@@ -156,6 +156,60 @@ It is therefore mandatory to use `CompletableFuture` APIs that are invoked even
 Failing to do so may result in the `Handler` `callback` parameter to never be completed, causing the request processing to hang forever.
 ====
 
+[[pg-server-http-handler-impl-request-listeners]]
+====== `Request` Listeners
+
+Application may add listeners to the `Request` object to be notified of particular events happening during the request/response processing.
+
+`Request.addIdleTimeoutListener(Predicate<TimeoutException>)` allows you to add an idle timeout listener, which is invoked when an idle timeout period elapses during the request/response processing, if the idle timeout event is not notified otherwise.
+
+The idle timeout listener is invoked only when:
+
+* There is no pending demand for request content; that is, there is not an outstanding call to `Request.demand(Runnable)`.
+* There is no pending response write; that is, there is not an outstanding call to `Response.write(boolean, ByteBuffer, Callback)`.
+
+In the above cases, the application is notified of the idle timeout so there is no need to call the idle timeout listeners.
+In the first case, the demand `Runnable` is invoked and the application will see the idle timeout failure by reading from the `Request`, obtaining a xref:pg-arch-io-content-source[transient failure chunk].
+In the second case, the response write `Callback` will be failed.
+
+The idle timeout listeners are therefore invoked only when the application is really idle, neither trying to read nor trying to write.
+
+An idle timeout listener may return `false` to indicate to the Jetty implementation that the idle timeout should be ignored, and `true` to indicate that the idle timeout should be treated as a fatal failure of the request/response processing.
```

Comment:
It is not really ignored, as the app might handle it.  How about:
```suggestion
An idle timeout listener may return `true` to indicate that the idle timeout should be treated as a fatal failure of the request/response processing; otherwise the listener may return `false` to indicate that no further handling of the timeout  is needed from Jetty.
```